app/Http/Controllers/UserController.php

```diff
@@ -63,29 +63,23 @@ public function store(Request $request): RedirectResponse
     /**
      * show user detail
      *
-     * @param integer $id
+     * @param User $user
      * @return mixed
      */
-    public function show(int $id): mixed
+    public function show(User $user): view
     {
-        $data = $this->userService->getUserById($id);
-        if ($data != null) {
-            return view('users.detail', ['user' => $data]);
-        } else {
-            return redirect()->route('users.index')->with('failed', 'User Does Not Exist');
-        }
+        return view('users.detail', ['user' => $user]);
     }
 
     /**
      * Show the form for editing the specified resource.
      *
-     * @param integer $id
+     * @param User $user
      * @return View
      */
-    public function edit(int $id): View
+    public function edit(User $user): View
```

Comment:
Same here. May be parameter is user_id and parameter type is `int`

Reply:
https://laravel.com/docs/10.x/routing#route-model-binding

implemented as Model instead of Int so that it shows an 404 instead of an error.